test/unit-tests/api-gateway/validation/oauth2/oauthTokenValidatorTest.lua

```diff
@@ -0,0 +1,145 @@
+    UNKNOWN_ERROR = { error_code = "503010", message = "Could not validate the oauth token" }
+}
+
+beforeEach(function()
+
+    ngx.header = {}
+
+    ngx.config = {
+        debug = false
+    }
+
+    ngx.__time.doReturn = function()
+        return 123
+    end
+
+    ngx.shared = {
+        cachedOauthTokens = shared
+    }
+
+    ngx.__now.doReturn = function()
+        return os.time()
+    end
+
+    ngx.req.__start_time.doReturn = function()
+        return os.time() - 10
+    end
+
+    hasherMock.__hash.doReturn = function(self)
+        return "hashedToken"
+    end
+
+    ngx.var.oauth_host = "oauth-na1.adobelogin.com"
+end)
+
+test('checkResponseFromAuth: should return false for an invalid json', function()
+    local classUnderTest = require('api-gateway.validation.oauth2.oauthTokenValidator'):new()
+    local tokenValidity, error = classUnderTest:checkResponseFromAuth("invalid_json", "key")
+    assertEquals(tokenValidity, false)
+end)
+
+test('checkResponseFromAuth: should return true for a valid json', function()
+    local classUnderTest = require('api-gateway.validation.oauth2.oauthTokenValidator'):new()
+    local token = {
+        id = "1234", scope = "openid email profile", user_id = "21961FF44F97F8A10A490D36", expires_in = "86400000", client_id = "client_id_test_2", type = "access_token"
+    }
+
+    local body = {
+        valid = true, expires_at = 3400, token = safeCjson.encode(token)
+    }
+
+    local testJson = {
+        status = 200,
+        body = safeCjson.encode(body)
+    }
+
+    local tokenValidity = classUnderTest:checkResponseFromAuth(testJson, "key")
+    assertEquals(tokenValidity, true)
+end)
+
+test('validateOauthToken: should return 200 for a valid json', function()
+    ngx.var.authtoken = "eyJmZyI6IlJFRDJIUUE2NzdSRTc3UURBQUFBQUFON0FBPT09PT09IiwiYXMiOiJpbXMtbmExLXN0ZzEiLCJjIjoid3RsK2taUU01azltVVpBVW81NUgzQT09IiwidXNlcl9pZCI6IjE5NzA0MjJFNTdBM0IyQjg3RjAwMDEwMUBBZG9iZUlEIiwibW9pIjoiYjY2YTcyNzciLCJzY29wZSI6Im9wZW5pZCxBZG9iZUlELHJlYWRfb3JnYW5pemF0aW9ucyxhZGRpdGlvbmFsX2luZm8ucHJvamVjdGVkUHJvZHVjdENvbnRleHQsYWRkaXRpb25hbF9pbmZvLmpvYl9mdW5jdGlvbixzZXNzaW9uIiwiY3JlYXRlZF9hdCI6IjE0ODU1MDgwMjYxMTciLCJpZCI6IjE0ODU1MDgwMjYxMTctNzZkNjZiYzEtYjI3Mi00ZGJiLThkYjUtYjE1MmVhODNiNTQ1Iiwic3RhdGUiOiJ7XCJzZXNzaW9uXCI6XCJodHRwczovL2ltcy1uYTEtc3RnMS5hZG9iZWxvZ2luLmNvbS9pbXMvc2Vzc2lvbi92MS9ZalpoT0dVeE1XUXRZelJsTVMwME5XUTFMVGxpTWpJdFltRmpZMlUwTlRsaU5ERXhMUzB4T1Rjd05ESXlSVFUzUVROQ01rSTROMFl3TURBeE1ERkFRV1J2WW1WSlJBXCJ9IiwidHlwZSI6ImFjY2Vzc190b2tlbiIsImV4cGlyZXNfaW4iOiI4NjQwMDAwMCIsImNsaWVudF9pZCI6IlNpdGVDYXRhbHlzdDEifQ.bTWdkAiK3iH9l5XtXELBQCdZ0femwzdkzySNtNt2rqFlpR-ZsxNsE1_VKUAsMLDdaLsNpcKMNPYACP0wCtFmYBJvWVyqN9iJnf9lBsL6UOxRd5WZEHzDruL0GGJLykI8iiqaCJ9irNNESELbEr7vtOJb9hVrytIjm8anxukLK8RSeffbStZm9iqZki9_G-r4s8W6q8UlLy1-0dkoPJorq9oo1GWC0GwfyD7Xecn4z6JWS9NlFtvDGLvFoMiGMaopBaƒdDado_O1Hrgvf5l4IKE-MnsRQID1dWyK2uQ83We6FpuuJ0NGk1ceNMhkFhr7_k9RGhBswYADUKlBMsot_Adw"
```

Comment:
tokens can be anything please don't use this real like tokens

Reply:
Done.